src/Page.js

```diff
@@ -530,11 +517,38 @@ Page.prototype.removeFrontMatter = function (includedPage) {
   return $.html();
 };
 
+/**
+  * Inserts the page layout's header to the start of the page
+  * @param pageData a page with its front matter collected
+  */
+Page.prototype.insertHeaderFile = function (pageData) {
+  const { header } = this.frontMatter;
+  let headerFile;
+  if (header) {
+    headerFile = path.join(HEADERS_FOLDER_PATH, header);
+  } else {
+    headerFile = path.join(LAYOUT_FOLDER_PATH, this.frontMatter.layout, LAYOUT_HEADER);
+  }
+  const headerPath = path.join(this.rootPath, headerFile);
+  if (!fs.existsSync(headerPath)) {
+    return pageData;
+  }
+  // Retrieve Markdown file contents
+  const headerContent = fs.readFileSync(headerPath, 'utf8');
+  // Set header file as an includedFile
+  this.includedFiles[headerPath] = true;
+  // Map variables
+  const newBaseUrl = calculateNewBaseUrl(this.sourcePath, this.rootPath, this.baseUrlMap) || '';
+  const userDefinedVariables = this.userDefinedVariablesMap[path.join(this.rootPath, newBaseUrl)];
+  return `${nunjucks.renderString(headerContent, userDefinedVariables)}\n${pageData}`;
+};
+
+
 /**
  * Inserts the footer specified in front matter to the end of the page
  * @param pageData a page with its front matter collected
  */
-Page.prototype.insertFooter = function (pageData) {
+Page.prototype.insertFooterFile = function (pageData) {
```

Comment:
Was thinking if this can be refactored into a plugin? Similar to #714 ? Just a side thought though.

Reply:
Will take a look if it is feasible. 

Although I am thinking that specifying plugins is less user intuitive compared to the current implementation of specifying via `<frontmatter>` or using layouts.

Reply:
We already pass `frontMatter` to plugins right?